content/purgatorio/launch-and-mantain-the-system/monitoring-usage-and-behavior.md

```diff
@@ -1,5 +1,142 @@
+
+So let's Start!
+
+The first resource you should go through is the following: 
+
+- [How to Monitor Machine Learning Models](https://christophergs.com/machine%20learning/2020/03/14/how-to-monitor-machine-learning-models/)
+
+It's a very detailed and comprehensive blog post and it addresses these topics:
+
+- What makes ML systems monitoring hard?
+- How can we monitor the usage and behavior of the model? 
+- What are the key metrics to track and alert on?
+- What are the key principles for monitoring the ML system?
+
+Once you are done with the previous blog post, you can refer to the related chapter of the book [Machine Learning Engineering](http://www.mlebook.com/wiki/doku.php) for a more detailed guide (strongly recommended buy, but you can read it for free online).
+
+- [Chapter 8 - Model Serving and Monitoring](https://www.dropbox.com/s/nj23m5myng9gram/Chapter8.pdf?dl=0)
+
+See also:
+ - [There are two very different ways to deploy ML models, here’s both](https://towardsdatascience.com/there-are-two-very-different-ways-to-deploy-ml-models-heres-both-ce2e97c7b9b1)
+
+With these two resources, you should understand broadly the reasons and challenges of monitoring the Machine Learning models in production, and you should have reasonable strategies to tackle them. 
+
+As you know, the two main challenges regarding the model monitoring are:
+
+- How does the system behave?
+- How is the system used?
+
+Let's see some specific resources for these two challenges.
+
+## Monitoring the Behavior
+
+When monitoring the behavior of the ML model in production, you should consider many aspects:
+
+
+- **Data in:**
+
+    - Monitor whether the data you're processing looks like the data you trained on. ([data drift challenge](https://www.cmswire.com/digital-experience/data-drift-what-it-is-and-how-to-avoid-it/)) E.g., use simple (comparatively interpretable) distributional models to try to track whether data looks "sufficiently" similar.
+    
+    See:
+    - [Calculate Similarity — the most relevant Metrics in a Nutshell](https://towardsdatascience.com/calculate-similarity-the-most-relevant-metrics-in-a-nutshell-9a43564f533e)
+    - [How to measure similarity or dissimilarity between two data set?](https://www.researchgate.net/post/How_to_measure_similarity_or_dissimilarity_between_two_data_set)
+
+
+- **Addressing data drift over time:**
+    
+    - Is your data distribution non-stationary? I.e. are you expecting your model to degrade due to the data changing over time?
+
+    - If so you can do anomaly detection on the stream and track the fractions of anomalous data points over time.
+    
+    See:
+    - [A Gentle Introduction to Concept Drift in Machine Learning](https://machinelearningmastery.com/gentle-introduction-concept-drift-machine-learning/)
+    - [Addressing feature drift in data streams using iterative subset selection](https://dl.acm.org/doi/10.1145/3325061.3325063)
```

Comment:
You may also add this https://github.com/EricSchles/backtester (from the same author who wrote `drifter_ml`...)